docs/cqrs/pipeline/adding_custom_middlewares.md

```diff
@@ -0,0 +1,74 @@
+
+# Adding custom middlewares
+
+Custom middlewares can further augment the pipeline by employing the `UseMiddleware` method on `IApplicationBuilder`. To illustrate, consider the following basic example, which introduces a check if employee is blocked into the pipeline:
```

Comment:
Maybe it's worth to mention `CQRSRequestPayload` and `CQRSEndpointMetadata` objects that might be useful when writing middleware.